src/backend/dx12/src/lib.rs

```diff
@@ -731,11 +731,21 @@ impl hal::Instance for Instance {
                 name.to_string_lossy().into_owned()
             };
 
+            // Check for WARP driver to verify if software rendering is being used.
+            // Verified by checking the existence of the flag: DXGI_ADAPTER_FLAG_SOFTWARE.
+            let software_renderer = {
+                if (desc.Flags & dxgi::DXGI_ADAPTER_FLAG_SOFTWARE) != 0 {
+                    true
+                } else {
+                    false
+                }
+            };
+
             let info = hal::AdapterInfo {
                 name: device_name,
                 vendor: desc.VendorId as usize,
                 device: desc.DeviceId as usize,
-                software_rendering: false, // TODO: check for WARP adapter (software rasterizer)?
+                software_rendering: software_renderer,
```

Comment:
if you rename the variable, you could just do `software_rendering,` here. Alternative, move the expression here